### ISSUE TITLE
Missing sanity check to prevent "No roots in graph" runtime exception

### ISSUE THREAD

**Issue by kphaison** [stanfordnlp/CoreNLP#803]
The following sentence is causing a runtime exception when generating universal dependencies.

**"(who provided me with above information)."** - from a real clinical note
  
Review of the existing class shows that the sanity check (or equivalent) exists in most methods:

    /* If the graph doesn't have a root (most likely because
     * a parsing error, we can't match Semgrexes, so do
     * nothing. */
    if (sg.getRoots().isEmpty())
      return;

Currently missing in demoteQuantificationalModifiers(SemanticGraph sg)

Will review methods and see where else the check should occur for that class.

**Comment by kphaison** [comment]
Also good at:
  `private static void processSimple2WP(SemanticGraph sg, HashMap<String, HashSet<Integer>> bigrams)`

  `private static void processComplex2WP(SemanticGraph sg, HashMap<String, HashSet<Integer>> bigrams) `

  `private static void process3WP(SemanticGraph sg, HashMap<String, HashSet<Integer>> trigrams) 
`
 ` private static void processNames(SemanticGraph sg) `

besides:
 `demoteQuantificationalModifiers(SemanticGraph sg)`

**Comment by kphaison** [comment]
Reviewed the code stack and there is a check already in 
` private static void processMultiwordPreps(SemanticGraph sg) ` (lines 1456-1458)
which then call 3 of the 5 methods above (only caller)
```
private static void processSimple2WP(SemanticGraph sg, HashMap<String, HashSet<Integer>> bigrams)
private static void processComplex2WP(SemanticGraph sg, HashMap<String, HashSet<Integer>> bigrams)
private static void process3WP(SemanticGraph sg, HashMap<String, HashSet<Integer>> trigrams) 
```

So technical to patch we could add the check in processNames() and demoteQuantificationModifiers(); for long term and defensive code - we need all 5 checks in place.

**Comment by AngledLuffa** [comment]
Thank you for pointing this out!  Your code changes have been released in version 4.0.0
